src/services/reallocateCourtCaseToForce.ts

```diff
@@ -8,80 +8,110 @@ import insertNotes from "./insertNotes"
 import courtCasesByOrganisationUnitQuery from "./queries/courtCasesByOrganisationUnitQuery"
 import updateLockStatusToUnlocked from "./updateLockStatusToUnlocked"
 import UnlockReason from "types/UnlockReason"
+import AuditLogEvent from "@moj-bichard7-developers/bichard7-next-core/build/src/types/AuditLogEvent"
+import storeAuditLogEvents from "./storeAuditLogEvents"
+import getCourtCase from "./getCourtCase"
+import getAuditLogEvent from "@moj-bichard7-developers/bichard7-next-core/build/src/lib/auditLog/getAuditLogEvent"
 
 const reallocateCourtCaseToForce = async (
   dataSource: DataSource | EntityManager,
   courtCaseId: number,
   user: User,
   forceCode: string,
   note?: string
-): Promise<UpdateResult | Error> => {
+): Promise<void> => {
   // TODO:
-  // - Add audit log messages: Old bichard pushes messages to GENERAL_EVENT_QUEUE which goes into audit log
   // - Generate TRPR0028 if necessary
-  const updateResult = await dataSource.transaction(
-    "SERIALIZABLE",
-    async (entityManager): Promise<UpdateResult | Error> => {
-      const courtCaseRepository = entityManager.getRepository(CourtCase)
-      let query = courtCaseRepository.createQueryBuilder().update(CourtCase)
-      const newForceCode = `${forceCode}${DEFAULT_STATION_CODE}`
-      query.set({ orgForPoliceFilter: newForceCode })
-      query = courtCasesByOrganisationUnitQuery(query, user) as UpdateQueryBuilder<CourtCase>
-      query.andWhere("error_id = :id", { id: courtCaseId })
-
-      const amendResult = await amendCourtCase(entityManager, { forceOwner: forceCode }, courtCaseId, user)
-
-      if (isError(amendResult)) {
-        throw amendResult
+  // - Reset triggers on reallocate
+  await dataSource.transaction("SERIALIZABLE", async (entityManager): Promise<void> => {
+    const events: AuditLogEvent[] = []
+
+    const courtCase = await getCourtCase(entityManager, courtCaseId)
+
+    if (isError(courtCase)) {
+      throw courtCase
+    }
+
+    if (!courtCase) {
+      throw new Error("Failed to unlock: Case not found")
```

Comment:
I think this error message isn't accurate since we're failing to reallocate the case here, not unlock it